### PR TITLE
Add unit tests for com.hankcs.hanlp.utility.MathUtility

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
## 注意事项

* 这次修改没有引入第三方类库。
* 也没有修改JDK版本号
* 所有文本都是UTF-8编码
* 代码风格一致
* [ x] 我在此括号内输入x打钩，代表上述事项确认完毕。

Hi,

I've analysed your code base and noticed that `com.hankcs.hanlp.utility.MathUtility` in the `hanlp` module is not fully tested.

I've written some tests that cover this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        MathUtility.normalizeExp(predictionScores);
+
+        Assert.assertArrayEquals(expected, predictionScores, DELTA);
+    }
+
+    @Test
+    public void testNormalizeExpHashMap() {
+        HashMap<String, Double> predictionScores =
+            new HashMap<String, Double>();
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java:47]: You don't need the second` HashMap<String, Double>`. 
You can do it like this `HashMap<String, Double> predictionScores = new HashMap<>();`

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    @Test
+    public void testNormalizeExpDoubleArray() {
+        double[] predictionScores = {0, 1, 2, 3};
+        double[] expected = {
+            0.03205860328008499, 0.08714431874203257,
+            0.23688281808991013, 0.6439142598879724
+        };
+
+        MathUtility.normalizeExp(predictionScores);
+
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java:34]: remove space? i think.. not sure now.. maybe not.. i don't know anymore...

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package com.hankcs.hanlp.utility;
+
+import com.hankcs.hanlp.corpus.tag.Nature;
+import com.hankcs.hanlp.dictionary.CoreDictionary;
+import com.hankcs.hanlp.seg.common.Vertex;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.HashMap;
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java]: imports be alphabetical order 'J' comes before 'O'

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+public class MathUtilityTest {
+    static final double DELTA = 0.0;
+
+    @Test
+    public void testAverage() {
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java]: Order of tests are incorrect.. should start with `int sum` method

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    @Test
+    public void testSumInt() {
+        Assert.assertEquals(0, MathUtility.sum(new int[0]));
+        Assert.assertEquals(106, MathUtility.sum(new int[]{1, 32, 73}));
+    }
+
+    @Test
+    public void testCalculateWeight() {
+        Assert.assertEquals(0.10536051123919675, MathUtility.calculateWeight(
+            new Vertex("foo"), new Vertex("Bar")), DELTA);
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java]: first break after ',' then break after '('

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            55
+        );
+        Vertex vertex2 = new Vertex(
+            "foo",
+            new CoreDictionary.Attribute(natures, new int[]{-1}),
+            65678
+        );
+
+        Assert.assertEquals(2.1972155419463637, MathUtility.calculateWeight(
+            vertex1, vertex2), DELTA);
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java]: first break after ',' then break after '('

**Review by Braavos6 — CHANGES_REQUESTED** [review]

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java (hunk 1 above)
Comment (ThomasPerkins1123) [reply src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java:47]: You can usually but this is a java 7+ feature but unfortunately this project is running java 6

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            0.23688281808991013, 0.6439142598879724
+        };
+        MathUtility.normalizeExp(predictionScores);
+
+        Assert.assertArrayEquals(expected, predictionScores, DELTA);
+    }
+    
+    @Test
+    public void testCalculateWeight() {
+        Assert.assertEquals(0.10536051123919675,
```
Comment (Braavos6) [inline src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java]: move this assert to the bottom.. the objects create below doesn't seem to affect the result of this assert. Look nicer to have the asserts in one place.. I would put it below the other assert as that uses the objects :)

File: src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java (hunk 1 above)
Comment (Braavos6) [reply src/test/java/com/hankcs/hanlp/utility/MathUtilityTest.java:47]: ok :)

**Review by Braavos6 — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Review by Braavos6 — APPROVED** [review]

**Pull request merged**